hyperspy/tests/signal/test_2D_tools.py

```diff
@@ -129,6 +129,96 @@ def test_align_expand(self):
         assert np.all(d_al == self.aligned)
 
 
+@lazifyTestClass
+class TestGetSignal2DScale:
+
+    def test_default_scale(self):
+        s = hs.signals.Signal2D(np.ones((100, 50)))
+        x0, y0, x1, y1, length = 10., 10., 30., 10., 80.
+        scale0 = s._get_signal2d_scale(
+                x0=x0, y0=y0, x1=x1, y1=y1, length=length, scaled_input=True)
+        scale1 = s._get_signal2d_scale(
+                x0=x0, y0=y0, x1=x1, y1=y1, length=length, scaled_input=False)
+        # With the default scale (1), scale0 and scale1 have the same value
+        assert scale0 == scale1 == 4.
+
+    def test_non_one_scale(self):
+        s = hs.signals.Signal2D(np.ones((100, 50)))
+        sa = s.axes_manager.signal_axes
+        x0, y0, x1, y1, length = 10., 10., 30., 10., 80.
+        sa[0].scale, sa[1].scale = 0.1, 0.1
+        scale0 = s._get_signal2d_scale(
+                x0=x0, y0=y0, x1=x1, y1=y1, length=length, scaled_input=True)
+        scale1 = s._get_signal2d_scale(
+                x0=x0, y0=y0, x1=x1, y1=y1, length=length, scaled_input=False)
+        assert scale0 == 0.4
+        assert scale1 == 4
+
+    def test_diagonal(self):
+        length = (200**2 + 100**2)**0.5
+        s = hs.signals.Signal2D(np.ones((100, 200)))
+        sa = s.axes_manager.signal_axes
+        sa[0].scale, sa[1].scale = 0.5, 1.
+        scale0 = s._get_signal2d_scale(
+                x0=0, y0=0, x1=100, y1=100, length=length/2, scaled_input=True)
+        scale1 = s._get_signal2d_scale(
+                x0=0, y0=0, x1=200, y1=100, length=length, scaled_input=False)
+        assert scale0 == 0.5
+        assert scale1 == 1.
+
+
+@lazifyTestClass
```

Comment:
I think that the tests are not running in lazy mode, because in order for that to work, the signals have to be defined in the class constructor e.g.

```python

class TestCalibrate2D:

    def __init__(self):
        self.s = Signal2D()
````